docs/source/petsc-interface.rst

```diff
@@ -280,4 +305,87 @@ before going on to solve the system as before:
            ksp.solve(b, x)
 
 
+Accessing the PETSc mesh representation
+=======================================
+
+Under the hood, Firedrake uses PETSc's DMPlex unstructured mesh
+representation. It uses a hierarchical approach, where entities
+of different dimension are put on different levels of the
+hierarchy. The single tetrahedral element shown on the left below
+may be interpreted using the graph representation on the right.
+Entities of dimension zero (vertices) are shown at the top.
+Entities of dimension one (edges) are shown on the next level down.
+Entities of dimension two (faces) are shown on the penultimate
+level and the (dimension three) element itself is on the bottom
+level. Edges in the graph indicate which entities own/are owned
+by others.
+
+.. image:: _static/dmplex.png
+   :width: 75%
+   :align: center
+
+The DMPlex associated with a given ``mesh`` may be accessed via
+its ``topology_dm`` attribute:
+
+.. code-block:: python3
+
+    plex = mesh.topology_dm
+
+All entities in a DMPlex are given a unique number. The range
+of these numbers may be deduced using the method
+``plex.getDepthStratum``, whose only argument is the entity
+dimension sought. For example, 0 for vertices, 1 for edges, etc.
+The hierarchical DMPlex structure may be traversed using other
+methods, such as ``plex.getCone``, ``plex.getSupport`` and
+``plex.getTransitiveClosure``. See the `Firedrake DMPlex paper`_
+for details.
```

Comment:
Also link to the relevant part of the petsc manual? https://petsc.org/release/docs/manual/dmplex/